CONTRIBUTING.md

```diff
@@ -325,6 +326,111 @@ if you change the history (rebase, squash, amend), and use git push --force to
+Presenters are defined in individual classes/modules under `app/presenters/*`.
+The presenters are loaded into the app in the `app/presenters.rb` file, and if you add a presenter, it needs to be added to that file.
+The convention is `ClassName: 'filename'`.
+
+**Routes** in Sinatra are kind of like a combination of routes and controllers in Rails.
+Each route is like an individual Sinatra app that inherits from `Core` (found in `app/routes/core.rb`).
+In a route, you can specify endpoints (like `exercism.io/login`) and define how the app should respond (i.e. accept some credentials and log the user in or handle errors accordingly).
+Routes are defined in individual modules under `app/routes/*`.
+The routes are loaded into the app in the `app/routes.rb` file, and if you add a route, it needs to be added to that file.
+The convention is `ClassName: 'filename'`.
+Routes are unique in that you'll also have to update the main `app.rb` with the new route info.
+The convention is `Routes::ClassName`.
+
+**Views** are a collection of templates that create html.
+Templates are a way to embed Ruby in your views so that you can serve up information,
+iterate over items without repeating yourself (like creating tables of data), etc.
+The templating engine that Exercism uses is [ERB](https://en.wikipedia.org/wiki/ERuby).
+
+#### Client and Frontend (JavaScript, Custom Directives, Bootstrap, etc)
+
+The client side JS is a bit... spread out. There are two different JS related directories: `client/` and `frontend/`.
+Both have angular elements but `client/` is in JS and `frontend/` is mostly CoffeeScript.
+
+**Frontend** is where Exercism stores all of it's production client side code.
+Exercism doesn't have much JS (outside of bootstrap) but `frontend/` does handle comment threads on submissions, markdown preview on comments, submission code (expanding and contracting), etc. If you suspect that there is frontend wizardry happening and you haven't seen it in bootstrap, most likely it's in here.
+
+#### DB (Migrations)
+
+To create a migration you can use `rake db:generate:migration name=the_name_of_your_migration`.
+That will create a new migration in `db/migrate`.
+For help, look at the other migrations or at docs for ActiveRecord migrations.
+If you add, remove or rename database columns, you may have to update the seeds as well.
+Exercism pulls seed data from another repo found [here](https://github.com/exercism/seeds).
+The seed data provides placeholder information (like users, exercise submissions, comments, etc) to help with development.
+
+#### Lib (DB Configurations, Application Logic, Models, Rake Tasks)
+
+**Application Logic** can be anything that is used for the backend application (and, possibly, shared with the frontend as well) can live here.
+Maybe a library for GitHub OAuth, configurations for a Markdown parser and a syntax highlighter, etc.
+
+**Models** are scaffolds for how data is stored and what it can do.
```

Comment:
Add more info to models and maybe a bit about how they relate to the db and migrations.